mods/ra2/rules/allied-infantry.yaml

```diff
@@ -125,26 +125,64 @@ e1:
 		HP: 125
 	Mobile:
 		Speed: 31
+		RequiresCondition: !chronodisable && undeployed
 	Passenger:
 		PipType: Green
 	RevealsShroud:
 		Range: 5c0
-	AttackFrontal:
-		Voice: Attack
 	Armament@primary:
 		Weapon: M60
-		RequiresCondition: !rank-elite
+		RequiresCondition: !rank-elite && !deployed
 	Armament@elite:
 		Weapon: M60E
-		RequiresCondition: rank-elite
-	# Armament@deployed:
-	#	Weapon: para
-	#	UpgradeTypes: deployed
-	#	UpgradeMinEnabledLevel: 1
-	# Armament@elite-deployed:
-	#	Weapon: paraE
-	#	UpgradeTypes: eliteparaweapon
-	#	UpgradeMinEnabledLevel: 1
+		RequiresCondition: rank-elite && !deployed
+	Armament@deployed:
+		Weapon: para
+		Name: deployed
+		RequiresCondition: !rank-elite && deployed
+	Armament@elite-deployed:
+		Weapon: paraE
+		Name: deployed
+		RequiresCondition: rank-elite && deployed
+	GrantConditionOnDeploy:
+		DeployedCondition: deployed
+		UndeployedCondition: undeployed
+		CanDeployOnRamps: True
+		Facing: 160
+		RequiresCondition: !parachute
+	WithInfantryBody:
+		RequiresCondition: undeployed
+	WithMakeAnimation:
+		Sequence: deploy
+		BodyNames: inviso
+	Turreted:
+		RealignDelay: -1
+		TurnSpeed: 50
+		InitialFacing: 128
+	WithSpriteTurret@idle:
+		Sequence: deployed
+		RequiresCondition: deployed && !AnimateTurret
+	WithSpriteTurret@animated:
+		RequiresCondition: deployed && AnimateTurret
+		Sequence: shoot-deployed
+	WithSpriteBody:
+		Sequence: inviso
+		Name: inviso
+	AttackFrontal:
+		Voice: Attack
+		RequiresCondition: undeployed
+	AttackTurreted@deployed:
+		Armaments: deployed
+		Voice: Attack
+		RequiresCondition: deployed
+	GrantConditionOnAttack:
+		Condition: AnimateTurret
+		RevokeDelay: 10
+		ArmamentNames: deployed
+	RejectsOrders@deployment:
+		Reject: Move, EnterTransport, EnterTransports
+		RequiresCondition: deployed
+	-TakeCover: # TODO: needs to be conditional so we can disable when deployed
```

Comment:
Is there an upstream ticket for this?
Would be good to create one (and link it to this PR as a use-case) if not.

Reply:
I think there is but should check.

Reply:
https://github.com/OpenRA/OpenRA/issues/15167 Issue already mentions GIs.